stream.go

```diff
@@ -312,6 +316,11 @@ func (s *Stream) Close() error {
 	s.stateLock.Unlock()
 	return nil
 SEND_CLOSE:
+	if !closeStream && s.session.config.StreamCloseTimeout > 0 && s.closeTimer == nil {
+		s.closeTimer = time.AfterFunc(
+			s.session.config.StreamCloseTimeout, s.closeTimeout)
```

Comment:
This seems to create a goroutine for 5 minutes every time one side closes an established stream. Should we pass a context to `closeTimeout` that gets canceled when `closeStream = true`? It seems like that would remove cleanup goroutines for well behaved peers.

Reply:
Yeah this was something I missed and I'm adding in now. I don't think we need a context, we can just cancel the timer. 

note timers under the hood don't spawn any goroutines, they sit in a single thread sleeping on a heap.

Reply:
Pushed a fix for that.